server/camphoric/pricing.py

```diff
@@ -116,12 +118,16 @@ def calculate_price(registration, campers):
 
 
 def datetime_to_dict(d):
-    # add time if only date type
-    if isinstance(d, date):
+    # during registration process, this may be None
+    if not d:
+        d = timezone.now()
+
+    # add time if only date type not datetime
+    if isinstance(d, date) and "timestamp" not in dir(d):
         d = datetime.combine(d, time(0, 0, 0, 0))
 
     return {
-        "epoch": d.timestamp() if d else 0,
+        "epoch": math.floor(d.timestamp()),
         "year": d.year if d else 0,
         "month": d.month if d else 0,
         "day": d.day if d else 0,
```

Comment:
If you keep the `d = timezone.now()` default above, you can remove the `if d else 0` from these lines.